PascalTriangle.java

```diff
@@ -0,0 +1,42 @@
+// the first and last element in each row will contain 1. And to populate the remaining element in each row, we just need to take sum 
+// of elements from the elements from the row before the current row. 
+
+// Time Complexity : O(numRows ^ 2)
+// Space Complexity L O(numRow ^ 2)
```

Comment:
This algorithm using constant auxiliary space. We don't count the required output's space towards SC, do we?